library/spdm_responder_lib/libspdm_rsp_set_certificate.c

```diff
@@ -147,63 +147,84 @@ libspdm_return_t libspdm_get_response_set_certificate(libspdm_context_t *spdm_co
     root_cert_hash_size = libspdm_get_hash_size(
         spdm_context->connection_info.algorithm.base_hash_algo);
 
-    if (request_size < sizeof(spdm_set_certificate_request_t) +
-        sizeof(spdm_cert_chain_t) + root_cert_hash_size) {
-        return libspdm_generate_error_response(spdm_context,
-                                               SPDM_ERROR_CODE_INVALID_REQUEST, 0,
-                                               response_size, response);
-    }
+    if ((libspdm_get_connection_version(spdm_context) >= SPDM_MESSAGE_VERSION_13) &&
+        ((spdm_request->header.param1 & SPDM_SET_CERTIFICATE_REQUEST_ERASE) != 0)) {
+        /*the CertChain field shall be absent;the value of SetCertModel shall be zero*/
+        if ((request_size != sizeof(spdm_set_certificate_request_t)) ||
+            ((spdm_request->header.param1 & SPDM_SET_CERTIFICATE_REQUEST_SET_CERT_MODEL_MASK) !=
```

Comment:
This check makes sense but it is not in the specification. It probably should be.

Reply:
Thanks.  In the spec `Table 94 Set certificate request attributes`:
`if this bit is set, the CertChain field shall be absent and the value of SetCertModel shall be zero.`